src/java.base/share/classes/java/lang/reflect/AccessibleObject.java

```diff
@@ -773,11 +773,8 @@ private boolean slowVerifyAccess(Class<?> caller, Class<?> memberClass,
      */
     private static boolean printStackTraceWhenAccessFails() {
         if (!printStackPropertiesSet && VM.initLevel() >= 1) {
-            String s = GetPropertyAction.privilegedGetProperty(
-                    "sun.reflect.debugModuleAccessChecks");
-            if (s != null) {
-                printStackWhenAccessFails = !s.equalsIgnoreCase("false");
-            }
+            printStackWhenAccessFails = GetBooleanAction.
+                privilegedGetProperty("sun.reflect.debugModuleAccessChecks");
```

Comment:
Running with `-Dsun.reflect.debugModuleAccessChecks` should set printStackWhenAccessFails to true, not false.

Reply:
You are right. The old way maps the null string to true, and the new way maps it to false. I did not notice that. At this point, I see no value in making the "true".equals and "false".equals changes. Too much can break. I'll reverse these changes.

Reply:
This change still needs to be reversed.

Reply:
The change has been reversed.

We had an internal discussion about IntelliJ's unboxing recommendation, and decided to let unboxing be a matter of personal preference. Those changes have been reversed too.